generate.xml

```diff
@@ -234,14 +248,22 @@
     <matrix>
```

Comment:
We could stand to have fewer builds compiling `png` and `qrencode` since we need to test both with and without it. Should also test with one but not the other as well.